src/renderer/services/wallet/balances.ts

```
@@ -82,33 +82,63 @@ export const createBalancesService = ({
+  const BATCH_DELAY_MS = 100
+
+  /**
+   * Process balance reloads in batches with a delay between batches
+   * This prevents overwhelming network/APIs with too many concurrent requests
+   */
+  const processBatchedReloads = (reloadFunctions: Array<() => void>): void => {
+    if (reloadFunctions.length === 0) return
+
+    // Process first batch immediately
+    const firstBatch = reloadFunctions.slice(0, BALANCE_BATCH_SIZE)
+    firstBatch.forEach((fn) => fn())
```

Biome (2.3.14)

[error] This callback passed to forEach() iterable method should not return a value.

Either remove this return or remove the returned value.


(lint/suspicious/useIterableCallbackReturn)